enterprise_access/utils.py

```diff
@@ -285,3 +287,43 @@ def get_course_run_metadata_for_assignment(assignment, content_metadata):
 
     # For course-based assignments, return metadata for the advertised course run
     return get_advertised_course_run_metadata(content_metadata)
+
+
+def time_until(date_str: str, unit: str = 'seconds', exact: bool = False):
```

Comment:
Django has some nice utilities around parsing dates in a timezone-aware way, and so on:
https://docs.djangoproject.com/en/5.1/ref/utils/#django.utils.dateparse.parse_datetime
You could achieve what you're doing here with something like:
```python
from django.utils import dateparse, timezone 
delta = dateparse.parse_datetime(date_str) - timezone.now() # result will be a timedelta object
delta_seconds = delta.seconds
```